src/reconcile/rec_write.c

```diff
@@ -2004,16 +2004,24 @@ __rec_split_write_reuse(
     if (mod->rec_result != WT_PM_REC_MULTIBLOCK || mod->mod_multi_entries < r->multi_next)
         return (false);
 
+    /*
+     * It is possible that the generated checksum value can be the same from two different images.
+     * Use the time aggregate also along with size and checksum in the comparison to identify
+     * whether the new image is as same as the previously written block to reuse.
+     */
     multi_match = &mod->mod_multi[r->multi_next - 1];
-    if (multi_match->size != multi->size || multi_match->checksum != multi->checksum) {
+    if (multi_match->size != multi->size || multi_match->checksum != multi->checksum ||
+      WT_TIME_AGGREGATE_IS_EMPTY(&multi->addr.ta) ||
+      WT_TIME_AGGREGATE_IS_EMPTY(&multi_match->addr.ta) ||
+      !WT_TIME_AGGRTEGATES_EQUAL(&multi->addr.ta, &multi_match->addr.ta)) {
```

Comment:
In theory, I think it is still possible we have two pages with the same size, checksum, and time aggregates. Should we just remove this optimization to be safe?

Reply:
Yes, I agree that in theory, it is possible. By adding the time aggregate into the comparison we are making it a very very low chance that it can happen. 
@agorrod opinions?

Reply:
I think this is the right intermediate step. We should make the check stronger by layering in the timestamp checks, but schedule future work to make it more robust.